packages/arb-token-bridge-ui/src/hooks/useRedeemRetryable.ts

```diff
@@ -1,42 +1,39 @@
-import { useState } from 'react'
-import {
-  L1ToL2MessageWriter as IL1ToL2MessageWriter,
-  L1ToL2MessageStatus
-} from '@arbitrum/sdk'
-import { useAccount, useSigner } from 'wagmi'
-import { TransactionReceipt } from '@ethersproject/providers'
+import { useCallback, useState } from 'react'
+import { L1ToL2MessageStatus } from '@arbitrum/sdk'
+import { useSigner } from 'wagmi'
 import dayjs from 'dayjs'
+import { TransactionReceipt } from '@ethersproject/providers'
 
 import { DepositStatus, MergedTransaction } from '../state/app/state'
 import { getRetryableTicket } from '../util/RetryableUtils'
 import { trackEvent } from '../util/AnalyticsUtils'
 import { getNetworkName } from '../util/networks'
 import { isUserRejectedError } from '../util/isUserRejectedError'
 import { errorToast } from '../components/common/atoms/Toast'
-import { useNetworks } from './useNetworks'
-import { useNetworksRelationship } from './useNetworksRelationship'
+import { getProviderForChainId } from './useNetworks'
 import { useTransactionHistory } from './useTransactionHistory'
+import { Address } from '../util/AddressUtils'
 
 export type UseRedeemRetryableResult = {
-  redeem: (tx: MergedTransaction) => void
+  redeem: () => void
   isRedeeming: boolean
 }
 
-export function useRedeemRetryable(): UseRedeemRetryableResult {
-  const [networks] = useNetworks()
-  const { address } = useAccount()
-  const { childChain, parentChainProvider } = useNetworksRelationship(networks)
+export function useRedeemRetryable(
+  tx: MergedTransaction,
+  address: Address | undefined
+): UseRedeemRetryableResult {
+  const { data: signer } = useSigner({ chainId: tx.childChainId })
```

Comment:
Now users don't have to do 2 clicks to switch + redeem. It happens in the same flow, and that's why we have to pass the correct chain ID to the signer or otherwise it will be a signer for the old network (before the switch happened)

Reply:
Can we use `destinationChainId` here, since we know it's a deposit

```suggestion
  const { data: signer } = useSigner({ chainId: tx.destinationChainId })
```